### DISCUSSION TITLE
offload decoding overhead with ffmpeg

### DISCUSSION THREAD

**Discussion by haggercody** [blakeblackshear/frigate#1630] [General]
this might be a stupid question, but i'm wondering, since i'm unable to get intel quicksync working in my setup, would there be a benefit to trying to run another app on my PC that could access the hardware decoding and pass it into frigate?  I was thinking maybe i could do something with ffmpeg, like decode the rstp stream into another format (rtmp or even mjpeg?).  Any ideas, is this stupid?

**Comment by blakeblackshear** [comment]
Even if you could find a way to do that, frigate would need to re-encode the video to save mp4 files. This would end up using more than decoding.

**Reply by haggercody** [reply]
Good point. I guess with the new beta we have contestant saving to disk right? 

I think I’m just going to bite the bullet and move to unbuntu desktop and see how it goes there

I’ve setup two cameras and am pretty happy with the detect results. Tho seeing about 50 percent of my cpu there. I’m comparison I have three cams in blue iris running about 5 percent so I’m curious how much of this is the lack of hardware accel

**Reply by blakeblackshear** [reply]
For one, Blue iris isn't doing object detection. You would have to turn off detection to get a fair comparison.

**Reply by haggercody** [reply]
oh totally frigate is doing much more than blue iris, so im not complaining that it takes more cpu;  But probably worth turning off detect if that gives me a good comparison about the delta of the cost of handling the streams without hardware decoding--i tried to remove the detect role but that seems required, so i wasnt sure how to disable detection?

Side note, installing ubuntu dekstop.  I needed a good reason to buy a new ssd anyway.  Do you recommend going the HASS addon route or run frigate in a docker + integration if im looking to minimize resource consumption?

**Reply by blakeblackshear** [reply]
You would add this under the camera:

```yaml
detect:
  enabled: false
```

How are you planning to run Home Assistant?

**Reply by haggercody** [reply]
thanks will give that a whirl.  
I was leaning towards doing the supervised install, but ill take recommendations.  I'd like to still use the desktop for other apps if i can.

**Reply by blakeblackshear** [reply]
The addon is a fine choice with a supervised install. If you want more flexibility in running frigate,you may to run directly in docker instead and use the new proxy addon.

**Comment by haggercody** [comment]
hmm, yea not much drop when i turn off detect--i can confirm the inference time is listed as 10 in debug view (was 300s before).  CPU usage dropped maybe 10-15 percent.

**Reply by blakeblackshear** [reply]
That must be the hardware acceleration for decoding then. That and object detection (without a Coral) are the major drivers of CPU usage.

**Comment by haggercody** [comment]
did a test with blue iris and turned off hardware accel and it goes up to around 15 percent cpu (from usually 4-5 when run as a service).  So yea, it makes a dent.  Still frigate seems to run a bit more with detection off and no accel than iris, but I'm guessing some more tuning ill need to figure out.. maybe will see better results in ubuntu.

**Comment by haggercody** [comment]
@blakeblackshear setup unbuntu desktop today.. ok that wasn't so bad.  I made a mistake installing home assistant, i think i wasn't SU'd as root (not a lot of linux experience, you can tell).  Anyway i got past it and refreshed from my HASS snapshot.  confirmed with the intel_gpu_top command that the hardware decoding was working.  CPU usage is quite a bit down--maybe 20percent avg, where my win10 pc with blue iris could be 15. so thats not too terrible at all.. and i think some other threads were had noted some perf issues with the new release that might help more.

**Reply by jasonpstokes** [reply]
Offtopic (and likely too late anyway) but if you've installed Home Assistant Supervised, be aware that it isn't supported on Ubuntu and so it may refuse to update (itself and/or addons) at some point if you can't maintain everything it wants to see in the underlying OS.

**Reply by haggercody** [reply]
not too late i think thats a good point and i noticed it wasn't supported, but saw lots of folks had been doing it.  Maybe the next best setup is to do home assistant in a vm and frigate in a docker--then i get the HAOS but can ensure frigate can utilize hardware devices.
